src/MediaWiki/Search/Search.php

```diff
@@ -110,13 +118,30 @@ public function setDB( DatabaseBase $connection ) {
 		$this->fallbackSearch = null;
 	}
 
+
+	/**
+	 * @return \LoadBalancer
+	 */
+	protected function getLoadBalancer() {
+
```

Comment:
I would have thought you make this available to the `ApplicationFactory` as dedicated service hiding `MediaWikiServices` as much we can to avoid issues when things change again on the MW side.

The DI container would always return a `DBLoadBalancer` instance ( avoiding the `null` case) and internally makes the call to either `MediaWikiServices` or `LBFactory` but it is hidden from the consumer who calls `ApplicationFactory::getMediaWikiDbLoadBalancer`.